ngraph/core/src/atomic_guard.hpp

```diff
@@ -0,0 +1,27 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+#include <atomic>
+
+namespace ov {
+
+// The class AtomicGuard is an atomic wrapper that provides a convenient RAII-style mechanism for emulate mutex
+class AtomicGuard {
+public:
+    AtomicGuard(std::atomic_bool& b) : m_atomic(b) {
+        bool exp = false;
+        while (!m_atomic.compare_exchange_strong(exp, true, std::memory_order_seq_cst, std::memory_order_seq_cst)) {
```

Comment:
nit: std::memory_order_seq_cst, std::memory_order_seq_cst is not necessary here

Reply:
Fixed